Classes/MSAITelemetryManager.m

```diff
@@ -113,6 +118,26 @@ - (void)unregisterObservers {
   _sessionEndedObserver = nil;
 }
 
+#pragma mark - Common Properties
+
++ (void)setCommonProperties:(NSDictionary *)commonProperties {
+  [[self sharedManager] setCommonProperties:commonProperties];
+}
+
+- (void)setCommonProperties:(NSDictionary *)commonProperties {
+  dispatch_barrier_async(_commonPropertiesQueue, ^{
```

Comment:
This allows us to do many concurrent reads since that is not harmful while still blocking the queue for writes using `dispatch_barrier_async` :)